src/Transform/ONNX/LayerNameToDebugInfoPass.cpp

```diff
@@ -0,0 +1,56 @@
+#include "src/Pass/Passes.hpp"
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/Pass/Pass.h"
+
+#include "llvm/ADT/StringRef.h"
+
+#include <memory>
+
+using namespace mlir;
+using namespace llvm;
+
+namespace onnx_mlir {
+namespace {
+struct LayerNameToDebugInfoPass
+    : public PassWrapper<LayerNameToDebugInfoPass, OperationPass<>> {
+  MLIR_DEFINE_EXPLICIT_INTERNAL_INLINE_TYPE_ID(LayerNameToDebugInfoPass)
+
+  LayerNameToDebugInfoPass() = default;
+  LayerNameToDebugInfoPass(const LayerNameToDebugInfoPass &pass) = default;
+
+  [[nodiscard]] StringRef getArgument() const override {
+    return "onnx-layer-name-debug-info";
+  }
+
+  [[nodiscard]] StringRef getDescription() const override {
+    return "Inserts the ONNX layer name of each operation into the debug info "
+           "associated with it.";
+  }
+
+  void runOnOperation() final;
+};
+
+void LayerNameToDebugInfoPass::runOnOperation() {
+  // Check whether op has a onnx_node_name attribute and put that on the
+  // existing debug info.
+  Operation *op = getOperation();
+
+  op->walk([](Operation *nestedOp) {
+    if (auto layerName =
```

Comment:
Every operation has a default location, so there is nothing to set here.

Reply:
You can test this yourself: if you parse an MLIR file without debug locations and tell mlir-opt to print the debug info, it will have created a default location. I originally had a test for what you describe here, but in the end it was basically meaningless as there is no such thing as no debug info.

Reply:
@ljfitz

Reply:
Discussed offline. For onnx ops (except constants) that do not have a `onnx_node_name`, we will add `INVALID:FXML-1477:<seq-no>` as the location.